src/main/java/com/example/onlinebookshop/services/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,113 @@
+package com.example.onlinebookshop.services.impl;
+
+import com.example.onlinebookshop.dto.cartitem.request.AddCartItemDto;
+import com.example.onlinebookshop.dto.cartitem.request.UpdateItemQuantityDto;
+import com.example.onlinebookshop.dto.shoppingcart.response.ShoppingCartDto;
+import com.example.onlinebookshop.entities.CartItem;
+import com.example.onlinebookshop.entities.ShoppingCart;
+import com.example.onlinebookshop.exceptions.CartItemAlreadyExistsException;
+import com.example.onlinebookshop.exceptions.EntityNotFoundException;
+import com.example.onlinebookshop.mapper.CartItemMapper;
+import com.example.onlinebookshop.mapper.ShoppingCartMapper;
+import com.example.onlinebookshop.repositories.book.BookRepository;
+import com.example.onlinebookshop.repositories.cartitem.CartItemRepository;
+import com.example.onlinebookshop.repositories.shoppingcart.ShoppingCartRepository;
+import com.example.onlinebookshop.services.ShoppingCartService;
+import java.util.Comparator;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final CartItemMapper cartItemMapper;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final BookRepository bookRepository;
+
+    @Override
+    public ShoppingCartDto getShoppingCartByUserEmail(String email) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserEmail(email);
```

Comment:
It could be simpler and using userId preferable 
```suggestion
        return cartMapper.toDto(repository.findByUserId(userId));
```